nevergrad/optimization/optimizerlib.py

```diff
@@ -256,6 +257,82 @@ def __init__(
 CMA = ParametrizedCMA().set_name("CMA", register=True)
 DiagonalCMA = ParametrizedCMA(diagonal=True).set_name("DiagonalCMA", register=True)
 
+class _FCMA(base.Optimizer):
+
+    def __init__(
+            self,
+            parametrization: IntOrParameter,
+            budget: Optional[int] = None,
+            num_workers: int = 1,
+            scale: float = 1.0,
+            popsize: int = 0
+    ) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self._scale = scale
+        self._popsize = popsize if popsize > 0 else \
+                            max(self.num_workers, 4 + int(3 * np.log(self.dimension)))
+        self._es: Optional[cmaes.Cmaes] = None
+        # delay initialization to ease implementation of variants
+        self.listx: tp.List[ArrayLike] = []
+        self.listy: tp.List[float] = []
+        self.to_be_asked: tp.Deque[np.ndarray] = deque()
+
+    @property
+    def es(self) -> cmaes.Cmaes:
+        if self._es is None:
+            self._es = cmaes.Cmaes(x0=np.zeros(self.dimension, dtype=np.float),
+                                   input_sigma=self._scale,
+                                   popsize=self._popsize, randn=self._rng.randn)
+        return self._es
+
+    def _internal_ask(self) -> ArrayLike:
+        if not self.to_be_asked:
+            self.to_be_asked.extend(self.es.ask())
+        return self.to_be_asked.popleft()
+
+    def _internal_tell(self, x: ArrayLike, value: float) -> None:
+        self.listx += [x]
+        self.listy += [value]
+        if len(self.listx) >= self.es.popsize:
+            try:
+                self.es.tell(self.listy, self.listx)
+            except RuntimeError:
+                pass
+            else:
+                self.listx = []
+                self.listy = []
+
+    def _internal_provide_recommendation(self) -> ArrayLike:
+        if self._es is None:
+            raise RuntimeError("Either ask or tell method should have been called before")
+        if self.es.best_x is None:
+            return self.current_bests["pessimistic"].x
+        return self.es.best_x  # type: ignore
+
+
+class ParametrizedFCMA(base.ConfiguredOptimizer):
+    """FCMA-ES optimizer, wrapping external implementation: https://github.com/dietmarwo/fast-cma-es
+
+    Parameters
+    ----------
+    scale: float
+        scale of the search
+    popsize: int
+        population size, should be n * self.num_workers for int n >= 1.
+        default is max(self.num_workers, 4 + int(3 * np.log(self.dimension)))
+    """
+
+    # pylint: disable=unused-argument
+    def __init__(
+        self,
+        *,
+        scale: float = 0.3,
+        popsize: int = 0
+    ) -> None:
+        super().__init__(_FCMA, locals())
+
+FCMA = ParametrizedFCMA().set_name("FCMA", register=True)
```

Comment:
You'll need to run pytest locally so that it adds a record for the instances of optimizers you specified.
See [here](https://facebookresearch.github.io/nevergrad/contributing.html#how-to-test-it) for more details, and more generally the whole "contributing" section.

Reply:
Ok, will check out how to configure pytest